components/browser/engine-gecko-nightly/src/main/java/mozilla/components/browser/engine/gecko/media/GeckoMediaDelegate.kt

```diff
@@ -34,4 +35,30 @@ internal class GeckoMediaDelegate(
             engineSession.notifyObservers { onMediaRemoved(media) }
         }
     }
+
+    override fun onRecordingStatusChanged(
+        session: GeckoSession,
+        devices: Array<out GeckoSession.MediaDelegate.RecordingDevice>
+    ) {
+        val genericDevices = devices.map { it.toGeneric() }
+        engineSession.notifyObservers {
+            onRecordingStateChanged(genericDevices)
+        }
+    }
+}
+
+private fun GeckoSession.MediaDelegate.RecordingDevice.toGeneric(): RecordingDevice {
```

Comment:
nit: I think this function name doesn't convey what is doing. Maybe the [type alias](https://kotlinlang.org/docs/reference/type-aliases.html) feature could be helpful here.`GeckoSession.MediaDelegate.RecordingDevice`  could be named as`GeckoRecordingDevice`, this way this function could be named `toRecordingDevice`.

Reply:
👍

Reply:
:+1: